webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFact.scala

```diff
@@ -1924,14 +1924,21 @@ object NodeFactSerialisation {
     implicit val codecIpAddress:      JsonCodec[IpAddress]      = JsonCodec.string.transform(IpAddress(_), _.inet)
     implicit val codecNodeTimezone:   JsonCodec[NodeTimezone]   = DeriveJsonCodec.gen
     implicit val codecMachineUuid:    JsonCodec[MachineUuid]    = JsonCodec.string.transform[MachineUuid](MachineUuid(_), _.value)
-    implicit val codecMachineType:    JsonCodec[MachineType]    = JsonCodec.string.transform[MachineType](
-      _ match {
-        case UnknownMachineType.kind  => UnknownMachineType
-        case PhysicalMachineType.kind => PhysicalMachineType
-        case x                        => VirtualMachineType(VmType.parse(x).getOrElse(VmType.UnknownVmType))
-      },
-      _.kind
-    )
+    implicit val codecMachineType:    JsonCodec[MachineType]    = {
+      val encoder: JsonEncoder[MachineType] = JsonEncoder.string.contramap(_.kind)
+      // for compat before correction of https://issues.rudder.io/issues/24971, we need to keep
+      // "physicalMachine". "virtualMachine" wasn't directly used
+      val decoder: JsonDecoder[MachineType] = JsonDecoder.string.map {
+        case s =>
+          s.toLowerCase match {
+            case UnknownMachineType.kind                      => UnknownMachineType
+            case PhysicalMachineType.kind | "physicalMachine" => PhysicalMachineType
```

Comment:
compatibility would still be broken : `s.toLowerCase` may only be `physicalmachine` or the lowercased kind

Reply:
it's the same capitalization problem with the kind of unknown machine type which is still `unknownMachineType`...
Maybe it's the `.toLowerCase` that should be removed

Reply:
right, there's a missing toLowerCase / one too many